### PR TITLE
Update Demo 

### PR DESCRIPTION
**Pull request by stevendborrelli**
- update labels to match
- add provider manifest
- add function-auto-ready

### CODE REVIEW

**Review by jbw976 — APPROVED** [review]
was there more you wanted to still add to this PR @stevendborrelli? title still says WIP 😅

File: kcl/definition.yaml [hunk 1]
```diff
@@ -29,4 +29,13 @@ spec:
...
-                description: ID of this Network that will be included in its labels for other objects to discover easily.
\ No newline at end of file
+                description: ID of this Network that will be included in its labels for other objects to discover easily.
+              providerConfigName:
+                type: string
+                description: ProviderConfig to use to provision resources
+                default: default
+              region:
+                type: string
+                description: Region where there resources will be created
```
Comment (jbw976) [inline kcl/definition.yaml]:
```suggestion
                description: Region where the resources will be created
```

Comment (stevendborrelli) [reply kcl/definition.yaml]: Fixed

**Comment by stevendborrelli** [comment]
@jbw976 you can review now.

File: kcl/provider.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Install the ec2 provider, which contains CRDs and Controllers for EC2 resources
+# like VPCs and VM Instances.
+#
+# Installing this package will automatically pull in the upbound-family-aws provider
+# Upbound Marketplace link: https://marketplace.upbound.io/providers/upbound/provider-aws-s3/v1.1.1
```
Comment (jbw976) [inline kcl/provider.yaml:5]:
```suggestion
# Upbound Marketplace link: https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.0
```

File: kcl/README.md [hunk 3]
```diff
@@ -185,4 +186,140 @@ kind: Result
...
+* Apply the CompositeResourceDefinition (XRD) and Composition
+* Create your Claim which will trigger the Composition to create the Resources
+
+### Installing Crossplane
+
+Install Crossplane via the [Getting Started](https://docs.crossplane.io/v1.15/getting-started/provider-aws/) guide. For this example a minimum Crossplane version of 1.14.x is required.
+
+### Installing the EC2 Provider
+
+Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.1.1>. A copy of the manifest is in this directory.
```
Comment (jbw976) [inline kcl/README.md]:
```suggestion
Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.0>. A copy of the manifest is in this directory.
```

File: kcl/README.md [hunk 4]
```diff
@@ -185,4 +186,140 @@ kind: Result
...
+
+The major Cloud providers (AWS, Azure, GCP) are broken up in to "family" providers in order to reduce CRDs installed on the cluster, which is why there is a separate provider for the EC2 related resources.
+
+```shell
+kubectl apply -f provider.yaml
+```
+
+Verify that your Provider was installed Correctly. You should see that the `upbound-provider-family-aws` provider was automatically installed. This provider supplies the common `ProviderConfig` CRD.
```
Comment (jbw976) [inline kcl/README.md]:
```suggestion
Verify that your Provider was installed correctly. You should see that the `upbound-provider-family-aws` provider was automatically installed. This provider supplies the common `ProviderConfig` CRD.
```

File: kcl/README.md [hunk 5]
```diff
@@ -185,4 +186,140 @@ kind: Result
...
+
+```shell
+❯ kubectl get provider.pkg
+NAME                      INSTALLED   HEALTHY   PACKAGE                                          AGE
+upbound-provider-aws-ec2   True        True      xpkg.upbound.io/upbound/provider-aws-ec2:v1.2.0   9m
+upbound-provider-family-aws              True        True      xpkg.upbound.io/upbound/provider-family-aws:v1.2.0              10m
```
Comment (jbw976) [inline kcl/README.md]: is it easy to make these columns aligned? if not then don't worry 😉

File: kcl/README.md [hunk 6]
```diff
@@ -185,4 +186,140 @@ kind: Result
...
+function.pkg.crossplane.io/function-auto-ready created
+```
+
+Ensure the KCL functions are healthy:
+
+```console
+❯ kubectl get -f functions.yaml 
+NAME           INSTALLED   HEALTHY   PACKAGE                                                  AGE
+function-kcl   True        True      xpkg.upbound.io/crossplane-contrib/function-kcl:v0.2.0   114s
+function-auto-ready   True        True      xpkg.upbound.io/crossplane-contrib/function-auto-ready:v0.2.1   118s
```
Comment (jbw976) [inline kcl/README.md]: same here for these columns to be aligned

**Review by jbw976 — APPROVED** [review]
small nits/typos, but nothing major blocking! thanks for getting this working for live provisioning!! 💪

File: kcl/definition.yaml (hunk 1 above)
Comment (stevendborrelli) [reply kcl/definition.yaml]: fixed

File: kcl/README.md [hunk 7]
```diff
@@ -185,4 +186,141 @@ kind: Result
...
+* Apply the CompositeResourceDefinition (XRD) and Composition
+* Create your Claim which will trigger the Composition to create the Resources
+
+### Installing Crossplane
+
+Install Crossplane via the [Getting Started](https://docs.crossplane.io/v1.15/getting-started/provider-aws/) guide. For this example a minimum Crossplane version of 1.14.x is required.
+
+### Installing the EC2 Provider
+
+Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.1>. A copy of the manifest is in this directory.
```
Comment (jbw976) [inline kcl/README.md:209]:
```suggestion
Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.0>. A copy of the manifest is in this directory.
```

File: kcl/provider.yaml (hunk 2 above)
Comment (jbw976) [reply kcl/provider.yaml:5]: this one should still be updated as suggested, right? it's intended to point to the marketplace listing of the provider in this file? (which is `provider-aws-ec2:v1.2.0`)

File: kcl/README.md (hunk 7 above)
Comment (jbw976) [reply kcl/README.md:209]: the link as is gives me a 404

**Pull request merged**